src/components/Form/Range/Range.js

```diff
@@ -1,82 +1,101 @@
+import { sortRangeAction } from '../../../actions/filterActions';
+import { useDefaultRange } from '../../../hooks/useDefaultRange';
 
-const Range = ({ label, min, max, step, rangeValue }) => {
-  const [minValue, setMinValue] = useState(min);
-  const [maxValue, setMaxValue] = useState(max);
+import classes from './Range.module.scss'
 
-  const htmlFor = `${label}-${Math.random()}`;
+const Range = ({ category, sortRangeAction, range }) => {
+  const defaultRange = useDefaultRange();
+  const [rangeValues, setRangeValues] = useState(null);
+  const htmlFor = `${category}-${Math.random()}`;
 
   useEffect(() => {
-    setMinValue(min);
-    setMaxValue(max);
-  }, [min, max]);
+    if (defaultRange) {
+      setRangeValues(defaultRange);
+      sortRangeAction(defaultRange);
+    }
+  }, [defaultRange]);
 
-  const getMin = (e) => {
-    setMinValue(e.target.value);
-  };
+  useEffect(() => {
+    if (!range) {
+      setRangeValues(defaultRange);
+      sortRangeAction(defaultRange);
+    }
+  }, [range]);
 
-  const getMax = (e) => {
-    setMaxValue(e.target.value);
+  const getValue = ({target: {name, value}}) => {
+    setRangeValues({ ...rangeValues, [name]: value });
   };
 
   const applyRange = () => {
-    rangeValue({minRange: minValue, maxRange: maxValue})
+    sortRangeAction(rangeValues)
   };
 
   return (
-    <div className={classes.range}>
-      <div className={classes.rangeItem}>
-        <label htmlFor={htmlFor}>Min {label}</label>
-        <input
-          value={minValue}
-          min={min}
-          max={max}
-          step={step}
-          type="number"
-          id={htmlFor}
-          onChange={getMin}
-        />
-      </div>
+    <Fragment>
+      {rangeValues && (
+        <div className={classes.range}>
+          <div className={classes.rangeItem}>
+            <label htmlFor={htmlFor}>Min {category}</label>
+            <input
+              name="minRange"
+              value={rangeValues.minRange}
+              min={defaultRange.minRange}
+              max={defaultRange.maxRange}
+              step={defaultRange.step}
+              type="number"
+              id={htmlFor}
+              onChange={getValue}
+            />
+          </div>
 
-      <div className={classes.rangeItem}>
-        <label htmlFor={htmlFor}>Max {label}</label>
-        <input
-          value={maxValue}
-          min={min}
-          max={max}
-          step={step}
-          type="number"
-          id={htmlFor}
-          onChange={getMax}
-        />
-      </div>
+          <div className={classes.rangeItem}>
+            <label htmlFor={htmlFor}>Max {category}</label>
+            <input
+              name="maxRange"
+              value={rangeValues.maxRange}
+              min={defaultRange.minRange}
+              max={defaultRange.maxRange}
+              step={defaultRange.step}
+              type="number"
+              id={htmlFor}
+              onChange={getValue}
+            />
+          </div>
```

Comment:
repeatable code, extract to shared input component

Reply:
Done